RelVal/ReleaseValidation.C

```diff
@@ -216,18 +248,56 @@ void ProcessMonitorObjectCollection(
     } else if (clname.BeginsWith("TH")) {
       TObject *o = (TObject *)o2MonObjColl->FindObject(olname.Data());
       WriteHisto(o);
+      Counts++;
        }else if(clname.BeginsWith("TProfile")){
       TObject* o=(TObject*)o2MonObjColl->FindObject(olname.Data());
        WriteProfile(o);
+       Counts++;
     } else if(clname.BeginsWith("TEfficiency")){
       TObject* o=(TObject*)o2MonObjColl->FindObject(olname.Data());
        WriteTEfficiency(o);
+       Counts++;
     }
   }
   printf("%d objects processed \n", Counts);
   return;
 }
 
+void ProcessDirCollection(TDirectoryFile *dirCollect) {
+  TString dirname = dirCollect->GetName();
+  printf("--- Process objects in the TDirectory %s ---\n", dirname.Data());
+  dirCollect->ls();
+  int nkeys = dirCollect->GetNkeys();
+  TList *lkeys = dirCollect->GetListOfKeys();
+  int Counts = 0;
+  for (int j = 0; j < nkeys; j++) {
+    TKey *k = (TKey *)lkeys->At(j);
+    TString clname = k->GetClassName();
+    TString olname = k->GetName();
+    printf("****** %s (class %s)   ******\n", olname.Data(), clname.Data());
+    if (clname == "TDirectoryFile") {
+      TDirectoryFile *dirObj =
+	(TDirectoryFile *)dirCollect->Get(olname.Data());
+      dirObj->ls();
+      ProcessDirCollection(dirObj);  
+    } else if (clname.BeginsWith("TH")) {
+      printf("--- Process histograms in %s ---\n", dirname.Data());
+      printf("--- %s ---\n", olname.Data());
+      TObject *o = (TObject *)dirCollect->Get(olname.Data());
```

Comment:
When you do the `->Get(...)` it should be  a `TObject` by default, right? I guess in that case we can omit the cast to `TObject` here